README.md

```diff
@@ -4,13 +4,104 @@ This is a [Singer](https://singer.io) tap that produces JSON-formatted data
 following the [Singer
 spec](https://github.com/singer-io/getting-started/blob/master/SPEC.md).
 
-This tap:
+## Installation
 
-- Pulls raw data from [FIXME](http://example.com)
-- Extracts the following resources:
-  - [FIXME](http://example.com)
-- Outputs the schema for each resource
-- Incrementally pulls data based on the input state
+```bash
+$ mkvirtualenv -p python3 tap-stripe
+$ pip install tap-stripe
+$ tap-stripe --config config.json --discover
+$ tap-stripe --config config.json --properties properties.json --state state.json
+```
+
+## Usage
+
+This section dives into basic usage of `tap-stripe` by walking through extracting
+data from the api.
+
+### Install
+
+```bash
+$ mkvirtualenv -p python3 tap-stripe
+$ pip install tap-stripe
+```
+
+or
+
+```bash
+$ git clone git@github.com:singer-io/tap-stripe.git
+$ cd tap-stripe
+$ mkvirtualenv -p python3 tap-stripe
+$ python install .
+```
+
+### Create the configuration file
+
+Create a config file containing the stripe credentials, e.g.:
+
+```json
+{
+  "client_secret": "sk_live_xxxxxxxxxxxxxxxxxxxxxxxx",
+  "account_id": "acct_xxxxxxxxxxxxxxxx",
+  "start_date": "2017-01-01T00:00:00Z"
+}
+```
+
+### Discovery mode
+
+The tap can be invoked in discovery mode to find the available stripe entities.
+
+```bash
+$ tap-stripe --config config.json --discover
+
+```
+
+A discovered catalog is output, with a JSON-schema description of each table. A
+source table directly corresponds to a Singer stream.
+
+### Field selection
+
+In sync mode, `tap-stripe` consumes the catalog and looks for streams that have been
+marked as _selected_ in their associated metadata entries.
+
+Redirect output from the tap's discovery mode to a file so that it can be
+modified:
+
+```bash
+$ tap-stripe -c config.json --discover > properties.json
+```
+
+Then edit `properties.json` to make selections. The stream's metadata entry (associated
+with `"breadcrumb": []`) gets a top-level `selected` flag, as does its columns' metadata
+entries.
+
+```diff
+[
+  {
+    "breadcrumb": [],
+    "metadata": {
+      "valid-replication-keys": [
+        "created"
+      ],
+      "table-key-properties": [
+        "id"
+      ],
+      "forced-replication-method": "INCREMENTAL",
++      "selected": "true"
+    }
+  },
+]
+```
+
+### Sync mode
+
+With a properties catalog that describes field and table selections, the tap can be invoked in sync mode:
+
+```bash
+$ tap-stripe -c config.json --properties properties.json
```

Comment:
Can we use `--catalog catalog.json` here since `--properties` is the legacy way of doing this?